spec/mongoid_search_spec.rb

```diff
@@ -19,11 +19,11 @@
     Mongoid::Search.stem_proc     = @default_proc
     @product = Product.create :brand => "Apple",
       :name => "iPhone",
-      :tags => (@tags = ["Amazing", "Awesome", "Olé"].map { |tag| Tag.new(:name => tag) }),
       :category => Category.new(:name => "Mobile", :description => "Reviews"),
       :subproducts => [Subproduct.new(:brand => "Apple", :name => "Craddle")],
       :info => { :summary => "Info-summary",
                  :description => "Info-description"}
+      @tags = ["Amazing", "Awesome", "Olé"].map { |tag| Tag.create(:name => tag, product: @product) }
```

Comment:
I suppose, you've changed tags creation because of ActiveSupport's `call` error, but if you use release Mongoid 6.0, this error doesn't appear. Revert tags initiation, it should work well

Reply:
@rtrv But, I am getting the same error if I run specs against MongoDB `2.6.12` and mongoid `6.0.3`. Here is the spec output

```
1) Mongoid::Search Serialized hash fields when the hash is populated should return the product
     Failure/Error: @product = Product.create :brand => "Apple",
     NoMethodError:
       undefined method `call' for #<ActiveSupport::Callbacks::CallbackSequence:0x007fd6906e8e08>
     # ./spec/mongoid_search_spec.rb:20:in `block (2 levels) in <top (required)>'
```

Reply:
@sivagollapalli it's strange, I've just fixed it by changing Mongoid 6 version from beta to release one. Try `6.0.0` instead of `6.0.3`

Reply:
@rtrv Just checked with mongoid `6.0.0`. Still, it shows same error.

Reply:
@sivagollapalli hm, interesting. I'm checking it on my machine at this week, wanna fix 6.0 support and some minor bugs to push the gem to 0.4 by the middle of December